lib/wincred/wincred.rb

```diff
@@ -77,8 +76,6 @@ def write_credential(
       write_result = Native.CredWriteW(Fiddle::Pointer[cred], 0)
 
       raise "Write to WinCred failed. Error code: #{Native.GetLastError}" if write_result.zero?
-    ensure
-      Native.CredFree(cred) if cred
```

Comment:
I think you ought to call `Fiddle.free` here instead.
As discussed, unfortunately it doesn't seem to clean it up on its own.